engine/runtime-compiler/src/main/scala/org/enso/compiler/pass/analyse/alias/Graph.scala

```diff
@@ -5,14 +5,17 @@ import org.enso.syntax.text.Debug
 import org.enso.compiler.pass.analyse.alias.Graph.{Occurrence, Scope}
 
 import java.util.UUID
+import scala.collection.immutable.HashMap
 import scala.collection.mutable
 import scala.reflect.ClassTag
 
 /** A graph containing aliasing information for a given root scope in Enso. */
 sealed class Graph extends Serializable {
-  var rootScope: Graph.Scope = new Graph.Scope()
-  var links: Set[Graph.Link] = Set()
-  var nextIdCounter          = 0
+  var rootScope: Graph.Scope                              = new Graph.Scope()
+  private var links: Set[Graph.Link]                      = Set()
+  private var sourceLinks: Map[Graph.Id, Set[Graph.Link]] = new HashMap()
+  private var targetLinks: Map[Graph.Id, Set[Graph.Link]] = new HashMap()
```

Comment:
There is an implementation of serialization logic for the `Graph` https://github.com/enso-org/enso/blob/6665c22eb9d5d8b1fd16ead769acaa3c31dda57b/engine/runtime-compiler/src/main/java/org/enso/compiler/pass/analyse/PassPersistance.java#L153-L157

Please make sure that it works correctly with the newly added fields and probably update the serialization id

Reply:
Should work correctly as I'm reproducing `sourceLinks` and `targetLinks` from links when deserialized. I'm not serializing those fields as there is no need for that (`links` continues to be serialized).